### PR TITLE
Publish docs

### PR DESCRIPTION
**Pull request by bhelx**
This PR does a couple things:

* Adds the links to the reference docs (which come from the generator)
* Adds the example code in the jsdocs
* Commits the docs in the `docs` folder which will be exposed when i turn on github pages.

### CODE REVIEW

File: lib/recurly/resources/Transaction.js [hunk 1]
```diff
@@ -29,7 +29,7 @@
  * @prop {string} id - Transaction ID
  * @prop {InvoiceMini} invoice
  * @prop {string} ipAddressCountry - IP address's country
- * @prop {string} ipAddressV4 - IP address provided when the billing information was collected:  - When the customer enters billing information into the Recurly.JS or Hosted Payment Pages, Recurly records the IP address. - When the merchant enters billing information using the API, the merchant may provide an IP address. - When the merchant enters billing information using the UI, no IP address is recorded.
+ * @prop {string} ipAddressV4 - IP address provided when the billing information was collected:  - When the customer enters billing information into the Recurly.JS or Hosted Payment Pages, Recurly records the IP address. - When the merchant enters billing information using the API, the merchant may provide an IP address. - When the merchant enters billing information using the UI, no IP address is recorded.
```
Comment (aaron-suarez) [inline lib/recurly/resources/Transaction.js]: nit: whitespace

File: lib/recurly/resources/Transaction.js [hunk 2]
```diff
@@ -40,7 +40,7 @@
...
  * @prop {Array.<string>} subscriptionIds - If the transaction is charging or refunding for one or more subscriptions, these are their IDs.
  * @prop {boolean} success - Did this transaction complete successfully?
- * @prop {string} type - - `authorization` – verifies billing information and places a hold on money in the customer's account. - `capture` – captures funds held by an authorization and completes a purchase. - `purchase` – combines the authorization and capture in one transaction. - `refund` – returns all or a portion of the money collected in a previous transaction to the customer. - `verify` – a $0 or $1 transaction used to verify billing information which is immediately voided.
+ * @prop {string} type - - `authorization` – verifies billing information and places a hold on money in the customer's account. - `capture` – captures funds held by an authorization and completes a purchase. - `purchase` – combines the authorization and capture in one transaction. - `refund` – returns all or a portion of the money collected in a previous transaction to the customer. - `verify` – a $0 or $1 transaction used to verify billing information which is immediately voided.
```
Comment (aaron-suarez) [inline lib/recurly/resources/Transaction.js]: here too

File: lib/recurly/resources/Settings.js [hunk 3]
```diff
@@ -10,7 +10,7 @@
  * Settings
  * @typedef {Object} Settings
  * @prop {Array.<string>} acceptedCurrencies
- * @prop {string} billingAddressRequirement - - full:      Full Address (Street, City, State, Postal Code and Country) - streetzip: Street and Postal Code only - zip:       Postal Code only - none:      No Address
+ * @prop {string} billingAddressRequirement - - full:      Full Address (Street, City, State, Postal Code and Country) - streetzip: Street and Postal Code only - zip:       Postal Code only - none:      No Address
```
Comment (aaron-suarez) [inline lib/recurly/resources/Settings.js]: nit: whitespace

File: lib/recurly/resources/Coupon.js [hunk 4]
```diff
@@ -15,7 +15,7 @@
  * @prop {string} couponType - Whether the coupon is "single_code" or "bulk". Bulk coupons will require a `unique_code_template` and will generate unique codes through the `/generate` endpoint.
  * @prop {datetime} createdAt - Created at
  * @prop {CouponDiscount} discount
- * @prop {string} duration - - "single_use" coupons applies to the first invoice only. - "temporal" coupons will apply to invoices for the duration determined by the `temporal_unit` and `temporal_amount` attributes.
+ * @prop {string} duration - - "single_use" coupons applies to the first invoice only. - "temporal" coupons will apply to invoices for the duration determined by the `temporal_unit` and `temporal_amount` attributes.
```
Comment (aaron-suarez) [inline lib/recurly/resources/Coupon.js]: nit: whitespace

File: README.md [hunk 5]
```diff
@@ -10,14 +10,8 @@ It's currently Beta software and is not yet an official release. Documentation f
...
 
-This library uses [documentation](http://documentation.js.org/) to generate jsdocs, but it's not published anywhere yet.
-To view documentation locally run:
-
-```
-./scripts/build && open docs/index.html
-```
-
-To view the HTTP API documentation along with the node example code, see [https://partner-docs.recurly.com/](https://partner-docs.recurly.com/).
+Docs and example code can be found here: [https://recurly.github.io/recurly-client-node](https://recurly.github.io/recurly-client-node).
```
Comment (aaron-suarez) [inline README.md]: This link 404s for me, what's the deployment plan here?

File: lib/recurly/resources/Coupon.js (hunk 4 above)
Comment (bhelx) [reply lib/recurly/resources/Coupon.js]: format should be taking care of this.

File: README.md (hunk 5 above)
Comment (bhelx) [reply README.md]: can't enable it until we merge this.

**Pull request merged**